core/src/main/java/hudson/model/Label.java

```diff
@@ -497,23 +497,27 @@ public Label paren() {
 
 
     @Override
-    public boolean equals(Object that) {
+    public final boolean equals(Object that) {
         if (this == that) return true;
         if (that == null || getClass() != that.getClass()) return false;
 
-        return name.equals(((Label)that).name);
+        return matches(((Label)that).name);
 
     }
 
     @Override
-    public int hashCode() {
+    public final int hashCode() {
         return name.hashCode();
     }
 
-    public int compareTo(Label that) {
+    public final int compareTo(Label that) {
         return this.name.compareTo(that.name);
     }
 
+    public final boolean matches(String labelString) {
```

Comment:
I would make this `private`, unless you plan to add `@since` and document exactly what it means. (From the name you might assume for example `Label.get("linux").matches("linux && x64")` or `Label.get("linux && x64").matches("linux")`, neither of which are true.)